yaml/src/Data/Yaml.hs

```diff
@@ -188,16 +189,20 @@ encodeFile = encodeFileWith defaultEncodeOptions
 -- @since 0.10.2.0
 encodeFileWith :: ToJSON a => EncodeOptions -> FilePath -> a -> IO ()
 encodeFileWith opts fp obj = runConduitRes
-    $ CL.sourceList (objToEvents opts $ toJSON obj)
+    $ CL.sourceList (objToStream opts obj)
    .| Y.encodeFileWith (encodeOptionsFormat opts) fp
 
-objToEvents :: EncodeOptions -> Value -> [Y.Event]
-objToEvents opts o = (:) EventStreamStart
+objToStream :: ToJSON a => EncodeOptions -> a -> [Y.Event]
+objToStream opts o = (:) EventStreamStart
               . (:) EventDocumentStart
-              $ objToEvents' o
+              $ objToEvents opts o ++
```

Comment:
I'd rather not have to lose the difference list approach here. It's not a major performance difference, but it's nice to keep.